plugins/modules/clickhouse_quota.py

```diff
@@ -0,0 +1,644 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2025, John Garland (@johnnyg) <johnnybg@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+---
+module: clickhouse_quota
+
+short_description: Creates or removes a ClickHouse quota.
+
+description:
+  - Creates or removes a ClickHouse quota.
+
+attributes:
+  check_mode:
+    description: Supports check_mode.
+    support: full
+
+version_added: '1.0.0'
```

Comment:
```suggestion
version_added: '1.1.0'
```
We follow semver and add new modules in minor releases, 1.1.0 is the next one